src/components/base/BaseButton.vue

```diff
@@ -0,0 +1,279 @@
+<script>
+import { defineComponent } from 'vue';
+import ArrowRightIcon from '@/assets/icons/arrow-right.svg';
+import ArrowLeftIcon from '@/assets/icons/arrow-left.svg';
+
+export default defineComponent({
+  components: {
+    ArrowRightIcon,
+    ArrowLeftIcon,
+  },
+  props: {
+    // TODO: add below line after implementation of ROUTING
+    // to: { type: [Object, String], default: null, required: false },
+    size: {
+      type: String,
+      default: 'regular',
+      validator(value) {
+        return ['small', 'medium', 'regular', 'large'].includes(value);
+      },
+    },
+    buttonType: {
+      type: String,
+      default: 'filled-red',
+      validator(value) {
+        return [
+          'filled-red',
+          'hollow-red',
+          'hollow-red-faded',
+          'filled-dark',
+          'hollow-dark',
+          'borderless-red',
+        ].includes(value);
+      },
+    },
+    arrowType: {
+      type: String,
+      default: 'none',
+      validator(value) {
+        return ['none', 'right', 'left'].includes(value);
+      },
+    },
+  },
+  computed: {
+    componentType() {
+      if (this.$attrs.href) return 'a';
```

Comment:
I like that you have prepared the button to be an `a` tag if needed and also prepared it for router introduction coming soon ❤️